src/core/mac/mac.cpp

```diff
@@ -1593,6 +1593,9 @@ Error Mac::ProcessReceiveSecurity(RxFrame &aFrame, const Address &aSrcAddr, Neig
         {
             aNeighbor->SetKeySequence(keySequence);
             aNeighbor->SetMleFrameCounter(0);
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+            aNeighbor->GetLinkFrameCounters().SetAll(frameCounter + 1);
```

Comment:
How about if we call `aNeighbor->GetLinkFrameCounter().Reset()`? (which makes it similar to what we do for with MLE Frame Counter?) I think this may be safer/better.

We know the `frameCounter + 1` is correct for the radio link on which we got this frame, but I think it is possible that `frameCounter + 1` is already ahead of the current frame counter being used by other radio types (on neighbor)?

Reply:
sure,  that would be cleaner approach, will incorporate the change. Do you think we can also remove the OPENTHREAD_CONFIG_MULTI_RADIO and do this in all scenarios, anyway for single radio case, it would update immediately, but I feel that would be more intuitive matching the spec description that all frame counters get reset on key sequence change?

Reply:
Thanks :)
Yes. Good idea to remove `OPENTHREAD_CONFIG_MULTI_RADIO` check and do it all cases.

Reply:
Thanks, made the suggested change